### PR TITLE
Bump dependencies to use Studio 4.0 and use  View Binding 

### PR DESCRIPTION
**Pull request by kriticalflare**
* Studio 4.0 is the latest stable android studio release, increase android gradle plugin version to be compatible with it 
* Use view binding to replace findViewById calls. Viewbinding is both type safe and null safe providing compile time safety.
* Move logic that operates on the view to onViewCreated according to [documentation](https://developer.android.com/reference/kotlin/androidx/fragment/app/Fragment?hl=en#onCreateView(android.view.LayoutInflater,%20android.view.ViewGroup,%20android.os.Bundle))
* Use placeholder string resources rather than concatenating strings together

### CODE REVIEW

**Comment by prabhuomkar** [comment]
Hey @kriticalflare I'd like if you create small PRs as we are building the base.
Let us know @ prabhuomkar@pm.me, dsouzadyn@gmail.com before you make breaking changes like this one :)

File: app/src/main/java/io/github/prabhuomkar/torchexpo/ui/about/AboutFragment.kt [hunk 1]
```diff
@@ -18,17 +17,21 @@ class AboutFragment : Fragment() {
     }
 
     private lateinit var viewModel: AboutViewModel
-    private lateinit var versionTextView: TextView
+    private var _binding: AboutFragmentBinding? = null
```
Comment (dsouzadyn) [inline app/src/main/java/io/github/prabhuomkar/torchexpo/ui/about/AboutFragment.kt:21]: Hey,
Is there a reason you aren't using LateInit here? We could get rid of the getter on the next line with LateInit.

File: app/src/main/java/io/github/prabhuomkar/torchexpo/ui/main/MainFragment.kt [hunk 2]
```diff
@@ -15,13 +15,16 @@ class MainFragment : Fragment() {
     }
 
     private lateinit var viewModel: MainViewModel
+    private var _binding: MainFragmentBinding? = null
```
Comment (dsouzadyn) [inline app/src/main/java/io/github/prabhuomkar/torchexpo/ui/main/MainFragment.kt:18]: We could use LateInit here too :)

**Comment by dsouzadyn** [comment]
@kriticalflare

File: app/src/main/java/io/github/prabhuomkar/torchexpo/ui/main/MainFragment.kt (hunk 2 above)
Comment (kriticalflare) [reply app/src/main/java/io/github/prabhuomkar/torchexpo/ui/main/MainFragment.kt:18]: Fragments can outlive their views , doing it this way scopes it to the lifecycle of the fragments view ( between onCreateView and onDestroyView ) . So  we also need to clean up any references to our views in onDestroyView.

This isn't the case in activities where lateinit suffices 
[View binding in activities](https://developer.android.com/topic/libraries/view-binding#activities) vs [View binding in fragments](https://developer.android.com/topic/libraries/view-binding#fragments)

**Comment by kriticalflare** [comment]
> Hey @kriticalflare I'd like if you create small PRs as we are building the base.
> Let us know @ prabhuomkar@pm.me, dsouzadyn@gmail.com before you make breaking changes like this one :)

Yeah I should have broken this into a smaller pr, will do next time

**Pull request merged**